spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SkipGramFeatureGenerator.java

```diff
@@ -0,0 +1,189 @@
+  private FeatureGeneratorConfig config;
+  private Schema outputSchema;
+  private Word2VecModel loadedModel;
+  private StructType schema;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+  }
+
+  @Override
+  public void initialize(SparkExecutionPluginContext context) throws Exception {
+    super.initialize(context);
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      throw new IllegalArgumentException(String.format("Failed to find model to use for Regression. Location does " +
+                                                         "not exist: %s.", modelLocation));
+    }
+    // load the model from a file in the model fileset
+    loadedModel = Word2VecModel.load(modelLocation.toURI().getPath());
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    if (input == null) {
+      return context.getSparkContext().emptyRDD();
+    }
+    JavaSparkContext javaSparkContext = context.getSparkContext();
+    SQLContext sqlContext = new SQLContext(javaSparkContext);
+    Schema inputSchema = input.first().getSchema();
+    List<StructField> structField = SparkUtils.getStructFieldList(inputSchema.getFields(),
+                                                                  config.getFeatureListMapping().keySet());
+    schema = new StructType(structField.toArray(new StructField[structField.size()]));
+    JavaRDD<Row> rowRDD = SparkUtils.convertJavaRddStructuredRecordToJavaRddRows(input, schema);
+    DataFrame documentDF = sqlContext.createDataFrame(rowRDD, schema);
+    DataFrame result = null;
+
+    for (Map.Entry<String, String> entry : config.getFeatureListMapping().entrySet()) {
+      result = loadedModel.setInputCol(entry.getKey() + "-transformed").setOutputCol(entry.getValue())
+        .transform(documentDF);
+      documentDF = result;
+    }
+
+    outputSchema = outputSchema != null ? outputSchema : config.getOutputSchema(inputSchema);
+
+    List<Column> requiredFields = new ArrayList<>();
+    for (Schema.Field field : outputSchema.getFields()) {
+      requiredFields.add(new Column(field.getName()));
+    }
+
+    JavaRDD<Row> transformedRDD =
+      javaSparkContext.parallelize(result.select(JavaConversions.asScalaBuffer(requiredFields).toSeq())
+                                     .collectAsList());
+    JavaRDD<StructuredRecord> output = transformedRDD.map(new Function<Row, StructuredRecord>() {
```

Comment:
``(mllib.feature.Word2VecModel)model.transform(word)``  ( in ``double[] wordVector = model.transform(word).toArray();``) requires the word to be present in the vocabulary of the model, otherwise the Word2VecModel throws IllegalStateException exception with message - `<word>` not in vocabulary.
Just wanted to confirm if this was the expected behaviour of the plugin. And, should we catch and ignore such words or throw error to the user?